deploy/templates/consumption-azuredeploy.json

```diff
@@ -351,74 +355,64 @@
+      "type": "Microsoft.EventHub/namespaces/eventhubs/consumergroups",
       "apiVersion": "2017-04-01",
-      "name": "[concat(variables('eventhub_namespace_name'), '/normalizeddata/reader')]",
+      "name": "[concat(variables('eventhub_namespace_name'), '/devicedata/$Default')]",
       "location": "[parameters('ResourceLocation')]",
       "dependsOn": [
         "[resourceId('Microsoft.EventHub/namespaces/eventhubs', variables('eventhub_namespace_name'), 'devicedata')]",
-        "[resourceId('Microsoft.EventHub/namespaces/eventhubs', variables('eventhub_namespace_name'), 'normalizeddata')]",
         "[resourceId('Microsoft.EventHub/namespaces', variables('eventhub_namespace_name'))]",
-        "[resourceId('Microsoft.EventHub/namespaces/AuthorizationRules', variables('eventhub_namespace_name'), 'RootManageSharedAccessKey')]"
+        "[resourceId('Microsoft.EventHub/namespaces/eventhubs/authorizationRules', variables('eventhub_namespace_name'), 'devicedata', 'writer')]",
+        "[resourceId('Microsoft.EventHub/namespaces/eventhubs/authorizationRules', variables('eventhub_namespace_name'), 'normalizeddata', 'reader')]"
       ],
       "properties": {
-        "rights": [
-          "Listen"
-        ]
       }
     },
     {
-      "type": "Microsoft.EventHub/namespaces/eventhubs/authorizationRules",
+      "type": "Microsoft.EventHub/namespaces/eventhubs/consumergroups",
       "apiVersion": "2017-04-01",
-      "name": "[concat(variables('eventhub_namespace_name'), '/normalizeddata/writer')]",
+      "name": "[concat(variables('eventhub_namespace_name'), '/normalizeddata/$Default')]",
       "location": "[parameters('ResourceLocation')]",
       "dependsOn": [
-        "[resourceId('Microsoft.EventHub/namespaces/eventhubs', variables('eventhub_namespace_name'), 'devicedata')]",
         "[resourceId('Microsoft.EventHub/namespaces/eventhubs', variables('eventhub_namespace_name'), 'normalizeddata')]",
         "[resourceId('Microsoft.EventHub/namespaces', variables('eventhub_namespace_name'))]",
-        "[resourceId('Microsoft.EventHub/namespaces/AuthorizationRules', variables('eventhub_namespace_name'), 'RootManageSharedAccessKey')]"
+        "[resourceId('Microsoft.EventHub/namespaces/eventhubs/authorizationRules', variables('eventhub_namespace_name'), 'devicedata', 'writer')]",
+        "[resourceId('Microsoft.EventHub/namespaces/eventhubs/authorizationRules', variables('eventhub_namespace_name'), 'normalizeddata', 'reader')]"
       ],
       "properties": {
-        "rights": [
-          "Send"
-        ]
       }
     },
     {
-      "type": "Microsoft.EventHub/namespaces/eventhubs/consumergroups",
-      "apiVersion": "2017-04-01",
-      "name": "[concat(variables('eventhub_namespace_name'), '/devicedata/$Default')]",
-      "location": "[parameters('ResourceLocation')]",
+      "type": "Microsoft.EventHub/namespaces/eventhubs/providers/roleAssignments",
+      "apiVersion": "2018-01-01-preview",
```

Comment:
Unfortunately, the best way to limit access to the Event Hub is to use a preview API. There is a way to avoid using a preview API, but the access control is less granular.